### ISSUE TITLE
Add paywall support for Facebook Instant Articles

### ISSUE THREAD

**Issue by tomasstark** [laterpay/laterpay-wordpress-plugin#1419]
We'd like to add support for [Facebook Instant Articles](https://www.facebook.com/formedia/solutions/instant-articles) to paywalled content. Prior to creating this task we did a discovery work and it seems to be possible in WordPress plugin only without requiring changes to our backend.

**User Story**
As the merchant, I want o be able to display my articles in FB Instant Articles so I can monetize my articles on the FB platform from my WordPress site.

---

**Acceptance Criteria**
- Update the plugin so that it will work with FB Instant articles:
  - FB Instant articles that are behind the paywall display properly in FB if the user has a subscription. Otherwise it will direct the user to Laterpay to subscribe.
  - Free articles remain free in FB Instant Articles.

Based on the discovery we completed, this will require the following:

- Create a test FB application with Instant Articles enabled
- Add logic to backend to pull subscriber's data on access check
- Implement FB Pixel to feed subscriber's info to Facebook so that next time they open the article it will be directly in Instant Articles platform without directing them to mobile version if they have access

---

**Discovery notes**
- Facebook Instant Articles docs: https://developers.facebook.com/docs/instant-articles
- Subscriptions documentation: https://developers.facebook.com/docs/instant-articles/subscriptions

- We're only going to support **Locked** and **Free** articles. Here are the [docs to Content Tiering](https://developers.facebook.com/docs/instant-articles/subscriptions/content-tiering) with these options described.

- It's not currently possible to support single article unlocks, so we're only going to support Subscriptions and Time Passes. We'll need to link those with FB users so they can read articles within IA platform without a paywall.

---

The flow works like this:

- Merchant sets up subscription offers on Facebook and fills out where users are redirected when they click offer from paywall. The mobile site then uses Facebook Pixel’s `<img>` tag to link visitors to their Facebook account and assigning them a valid subscription if they have one.

- All articles are pushed to Facebook and hosted by them on their servers. We push HTML of the article to them. Once article is pushed to FB, merchant can flag each article as either locked (paywalled) or free (for everyone, no paywall displayed). *Note that pushing articles to Facebook is not in scope. There is [official plugin](https://wordpress.org/plugins/fb-instant-articles/) that merchants can install and use for offloading their articles to Facebook.*

- Currently FB displays a paywall in their native app when visiting article from Android devices only. On iOS device you’re redirected to the mobile version of the site with a paywall. This means that iOS users would land on the article with Laterpay (not FB) paywall displayed. Android users would end up here too but they first need to click the link in FB’s paywall offer.

- Once a user lands on the mobile site and successfully authorises themselves to prove that they have access to the content, we use FB Pixel to create a reference of their subscription on Facebook. This is called [Account Linking](https://developers.facebook.com/docs/instant-articles/subscriptions/account-linking).

- FB expects us to set the timestamp that user’s subscription is valid through along with an identifier of the user on our systems. Once we have the reference to user's subscription and its length pushed to FB, next time this user opens article from FB app they won’t see the paywall and they can read articles within IA platform for the period of their subscription.

- When pushing the reference from FB Pixel, we can use Laterpay’s user token which we can hash with `md5` to keep it at a convenient length.

**Comment by ksmoore125** [comment]
`Won't Do` - As we are nearing the end of our hours with rtCamp, we will close this task here and save it for later using the Jira equivalent.
